azurelinuxagent/ga/exthandlers.py

```diff
@@ -871,6 +917,34 @@ def collect_ext_status(self, ext):
 
         return ext_status
     
+    def is_ext_handling_complete(self, ext):
```

Comment:
I think it would be more useful for debugging and logging to define a `__str__` method on ExtensionStatus and return the full status.